### DISCUSSION TITLE
version 3.2.2-BETA.1

### DISCUSSION THREAD

**Discussion by chinarui-na** [m1dugh/native-sound-mixer#11] [Q&A] [answered]
Now i use v3.2.2-BETA.1 with electron.
I package it.
Some computer can open and use it. But some computer open it with blank. The previous version is OK. Does latest version use senior api？ Whether it depends on the.NET version？

**Comment by m1dugh** [comment]
What does blank mean ? Is it not loaded by electron or the package just cannot be used ? I noticed a bug at some function in c++ that could lead to errors when retrieving session by its id, it should be soon fixed. The last pushed commits are 2-commits ahead of version 3.2.2-BETA.1 they might fix the problem for windows only as the linux new version hasn't been yet developed. It appears weird to me the new version doesn't work the same as the old since i only packaged old code into a class. Moreover this package relies on windows core audio api which is supposedly unchanged since windows 7. I'll dig more into that but since it works on my pc it's a bit hard to fix those errors 😅.

**Reply by chinarui-na** [reply]
Yes.blank is mean program crashes. Windows system logs are displayed：`Application - Specific permission Settings are not available to the application container. SID do not use...`.

In addition. My program uses native-sound-mixer, and if my program opens first, other audio programs open later. My program will not be able to capture this session. My code is:
```
import SoundMixer from "native-sound-mixer";
const devices = SoundMixer.devices;
    devices.forEach(item => {
      if (item.type == 0) {
        item.sessions.forEach(session => {
         console.log(session)
        })
      }
    })
```
By the way, it didn't appear in the previous version,such as v3.2.0 . but in the current version:v3.2.2-BETA.1

**Reply by m1dugh** [reply]
I ran into problems with linux compilations due to recent node-addon-api update, I'll try to fix this as soon as possible. Until I fix this issue, I suggest you go back to version `3.2.0`.

**Reply by chinarui-na** [reply]
Thanks, Waiting for fixed.

**Comment by m1dugh** [comment, marked as answer]
version `v3.2.2` is live and should fix all problems you have been having with the package ( some old codes might no longer work as ids has bee removed from both devices and AudioSessions )

**Reply by chinarui-na** [reply]
Now It works fine. Thanks.
